src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,250 @@
+    } else {
+      for (int j = 0; j < xdim - po; j++)
+        x(po + j) = x_block_padded(j);
+      for (int j = 0; j < M-1 && j < xdim - po; j++)
+        x(po + j) += temp_pad(j);
+    }
+  }
+}
+
+void RepeatedlyAdd(Vector<BaseFloat> &x, Vector<BaseFloat> &h) {
+  for (int po = 0; po < x.Dim(); po += h.Dim()) {
+    int remain = x.Dim() - po;
+    int l = h.Dim();
+    if ( remain < l) l = remain;
+    SubVector<float> x_block(x, po, l);
+    SubVector<float> subh(h, 0, l);
+    x_block.AddVec(1.0, subh);
+  }
+}
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+
+    const char *usage =
+        "Corrupts the wave files supplied via input pipe with the specified\n"
+        "room-impulse response (RIR) and additive noise distortions\n"
+        "(specified by corresponding files).\n";
+
+    ParseOptions po(usage);
+    std::string noise_file;
+    BaseFloat snr_db = 20;
+
+    po.Register("noise-file", &noise_file,
+                "file with additive noise");
+    po.Register("snr-db", &snr_db,
+                "desired SNR(dB) of the output");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() != 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    std::string source_file = po.GetArg(1);
+    std::string rir_file = po.GetArg(2);
+    std::string out_file = po.GetArg(3);
+
+    WaveData src_wave;
+    {
+      Input ki(source_file);
+      src_wave.Read(ki.Stream());
+    }
+    const Matrix<BaseFloat> &SRC = src_wave.Data();
+    BaseFloat samp_freq_src = src_wave.SampFreq();  // read sampling fequency
+    int32 num_samp_src = SRC.NumCols(),  // number of samples in recording
+          num_chan_src = SRC.NumRows();  // number of channels in recording
+
+    KALDI_LOG << "samp_freq of source is " << samp_freq_src
+              << " num_samp " << num_samp_src
+              << " num_chan " << num_chan_src;
+
+    WaveData rir_wave;
+    {
+      Input ki(rir_file);
+      rir_wave.Read(ki.Stream());
+    }
+    const Matrix<BaseFloat> &RIR = rir_wave.Data();
+    BaseFloat samp_freq_rir = rir_wave.SampFreq();  // read sampling fequency
+    int32 num_samp_rir = RIR.NumCols(),  // number of samples in recording
+          num_chan_rir = RIR.NumRows();  // number of channels in recording
+    KALDI_LOG << "samp_freq of rir is " << samp_freq_rir
+               << " num_samp " << num_samp_rir
+               << " num_chan " << num_chan_rir;
+
+    Vector<BaseFloat> src(num_samp_src);
+    src.CopyRowFromMat(SRC, 0);
+
+    Vector<BaseFloat> rir(num_samp_rir);
+    rir.CopyRowFromMat(RIR, 0);
+    // Assume the rir files are 32 bits
+    rir.Scale(1 / static_cast<float>(2147483647));
+
+    Vector<BaseFloat> noise(0);
+    if (!noise_file.empty()) {
+      float peak = 0;
+      int peak_index = 0;
+      peak = rir.Max(&peak_index);
+      KALDI_LOG << "peak is " << peak;
+      KALDI_LOG << "peak_index is " << peak_index;
+
+      int early_start = peak_index - 0.001 * samp_freq_rir;
```

Comment:
early_start => early_rir_start_index
early_end => early_rir_end_index

declare constants to store the values 0.001 and 0.05, to ensure clarity